### PR TITLE
ROX-20410: Add ScannerV4 Vulnerability Definitions Card

### PR DESCRIPTION
**Pull request by sachaudh**
## Description

This PR includes these main changes:
1. Rename the `Vulnerability definitions` card to `StackRox Scanner Vulnerability Definitions` to differentiate it from the new scanner
2. Add another card for `Scanner V4 Vulnerabilities` for the new scanner
3. Modify the API service function to explicitly state which component we need to query for when getting vulnerability definitions

## Screenshots

Screenshot 2024-01-23 at 3 20 38 PM

### CODE REVIEW

File: ui/apps/platform/src/Containers/SystemHealth/DashboardPage.js [hunk 1]
```diff
@@ -66,7 +70,18 @@ const SystemHealthDashboardPage = () => {
...
                     {hasReadAccessForAdministration && (
                         <GridItem span={12}>
-                            <VulnerabilityDefinitionsHealthCard pollingCount={pollingCountSlower} />
+                            <VulnerabilityDefinitionsHealthCard
+                                component="SCANNER"
+                                pollingCount={pollingCountSlower}
+                            />
+                        </GridItem>
+                    )}
+                    {hasAccessForScannerV4 && hasReadAccessForAdministration && (
```
Comment (sachaudh) [inline ui/apps/platform/src/Containers/SystemHealth/DashboardPage.js:79]: @RTann @dcaravel I just wanted to double-check if the logic for showing this card is correct. We will check if the feature flag is on and has `Administration` read access (same as above)

Comment (dcaravel) [reply ui/apps/platform/src/Containers/SystemHealth/DashboardPage.js:79]: That's accurate per my understanding, confirmed that the API to pull the vuln defs update requires `Administration` access:

https://github.com/stackrox/stackrox/blob/03b5f817f0087480b29c796db2b243d327281144/central/integrationhealth/service/service_impl.go#L28-L30

File: ui/apps/platform/src/Containers/SystemHealth/DashboardPage.js [hunk 2]
```diff
@@ -27,6 +28,9 @@ const SystemHealthDashboardPage = () => {
     const hasReadAccessForCluster = hasReadAccess('Cluster');
     const hasReadAccessForIntegration = hasReadAccess('Integration');
 
+    const { isFeatureFlagEnabled } = useFeatureFlags();
+    const hasAccessForScannerV4 = isFeatureFlagEnabled('ROX_SCANNER_V4');
```
Comment (dcaravel) [inline ui/apps/platform/src/Containers/SystemHealth/DashboardPage.js:32]: nit: the name of this variable may be misleading (unless this is a UI convention).  The feature flag is unrelated to 'access'.  Perhaps `isScannerV4Enabled` would be more accurate?

**Review by dcaravel — APPROVED** [review]
Changes LGTM from a Scanner V4 perspective. - one minor nit.  Recommend waiting for another review before merge.

**Review by vjwilson — APPROVED** [review]

**Pull request merged**